server/aesdsocket.c

```diff
@@ -194,11 +196,64 @@ void *socketThread(void *arg)
 			return NULL;
 		}
 		
+		// Handle ioctl
+		if (strncmp(recv_buf, "AESDCHAR_IOCSEEKTO:", 19) == 0) {
+		    unsigned int write_cmd, offset;
+			
+			if (sscanf(recv_buf, "AESDCHAR_IOCSEEKTO:%u,%u", &write_cmd, &offset) == 2) {
+			    struct aesd_seekto seekto;
+                seekto.write_cmd = write_cmd;
+                seekto.write_cmd_offset = offset;
+				
+				int iofd = open(filename, O_RDWR);
+		        if (iofd == -1){
+			        perror("open");
+			        syslog(LOG_ERR, "open");
+					pthread_mutex_unlock(&file_mutex);
+			        return NULL;
+		        }
+				
+                int result_ret = ioctl(iofd,AESDCHAR_IOCSEEKTO,&seekto);
+			    if (result_ret == -1){
+			        perror("ioctl");
+			        syslog(LOG_ERR, "ioctl");
+					close(iofd);
+					pthread_mutex_unlock(&file_mutex);
+			        return NULL;
+		        }
+				
+				while ((ret_byte = read(iofd, send_buf, sizeof(send_buf))) > 0){

```

Comment:
Shouldn't need to duplicate logic here with the read/send logic used when not sending ioctl